tests/publiccloud/aws_cli.pm

```diff
@@ -49,6 +49,9 @@ sub run {
     assert_script_run("aws ec2 describe-instances --filters 'Name=tag:openqa-cli-test-tag,Values=$job_id'", 90);
     my $instance_id = script_output("aws ec2 describe-instances --filters 'Name=tag:openqa-cli-test-tag,Values=$job_id' --output=text --query 'Reservations[*].Instances[*].InstanceId'", 90);
 
+    # Wait until the instance is really running
+    script_retry("aws ec2 describe-instances --instance-ids $instance_id --query 'Reservations[*].Instances[*].State.Name' --output text | grep 'running'", 90, delay => 15, retry => 12);
```

Comment:
isn't that something what we usually soling with `wait_for_ssh` ?

Reply:
I find this more elegant actually :) And the `wait_for_ssh` is a bit more complex than I want.

Reply:
Also the answer to your question is no as the problem is not while waiting for SSH but for waiting for AWS to assign public IP to the instance. Normally this is done by our Terraform but here I test the CLI tools where it isn't automated.